src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/NativeMemory.cs

```diff
@@ -0,0 +1,132 @@
+        /// <param name="byteCount">The size, in bytes, of the block to allocate.</param>
+        /// <returns>A pointer to the allocated aligned block of memory.</returns>
+        /// <exception cref="OutOfMemoryException">Allocating <paramref name="byteCount" /> of memory failed.</exception>
+        /// <remarks>This method is a thin wrapper over the C <c>aligned_alloc</c> API or a platform dependent aligned allocation API such as <c>_aligned_malloc</c> on Win32.</remarks>
+        [CLSCompliant(false)]
+        public static void* AlignedAlloc(nuint alignment, nuint byteCount)
+        {
+            void* result = aligned_alloc(alignment, byteCount);
+
+            if (result == null)
+            {
+                throw new OutOfMemoryException();
+            }
+
+            return result;
+        }
+
+        /// <summary>Frees an aligned block of memory.</summary>
+        /// <param name="ptr">A pointer to the aligned block of memory that should be freed.</param>
+        /// <remarks>
+        ///    <para>This method does nothing if <paramref name="ptr" /> is <c>null</c>.</para>
+        ///    <para>This method is a thin wrapper over the C <c>free</c> API or a platform dependent aligned free API such as <c>_aligned_free</c> on Win32.</para>
+        /// </remarks>
+        [CLSCompliant(false)]
+        public static void AlignedFree(void* ptr)
+        {
+            aligned_free(ptr);
+        }
+
+        /// <summary>Allocates a block of memory of the specified size, in bytes.</summary>
+        /// <param name="byteCount">The size, in bytes, of the block to allocate.</param>
+        /// <returns>A pointer to the allocated block of memory.</returns>
+        /// <exception cref="OutOfMemoryException">Allocating <paramref name="byteCount" /> of memory failed.</exception>
+        /// <remarks>This method is a thin wrapper over the C <c>malloc</c> API.</remarks>
+        [CLSCompliant(false)]
+        public static void* Alloc(nuint byteCount)
+        {
+            void* result = malloc(byteCount);
+
+            if (result == null)
+            {
+                throw new OutOfMemoryException();
+            }
+
+            return result;
+        }
+
+        /// <summary>Allocates a block of memory of the specified size, in elements.</summary>
+        /// <param name="elementCount">The count, in elements, of the block to allocate.</param>
+        /// <param name="elementSize">The size, in bytes, of each element in the allocation.</param>
+        /// <returns>A pointer to the allocated block of memory.</returns>
+        /// <exception cref="OutOfMemoryException">Allocating <paramref name="elementCount" /> * <paramref name="elementSize" /> bytes of memory failed.</exception>
+        /// <remarks>This method is a thin wrapper over the C <c>calloc</c> API.</remarks>
+        [CLSCompliant(false)]
+        public static void* AllocZeroInitialized(nuint elementCount, nuint elementSize)
```

Comment:
This notably isn't `Calloc` like approved, but rather the name suggested by Jan in the issue as a better alternative.

We should get "explicit" approval from API review before merging, but that can likely happen over e-mail.

Reply:
This set looks incomplete unless you are familiar with intricacies of C standard library. Should we have `void* AllocZeroInitialized(nuint byteCount)` and `void* Alloc(nuint elementCount, nuint elementSize)` as well?

Reply:
Potentially, but we'd likely need the same approval from API Review. The same would apply if we wanted `Try*` variants which don't throw.

Reply:
> AllocZeroInitialized

Just noticed that the equivalent API in Rust is called `alloc_zeroed`. Would it be a better name (since it is shorter)?

Reply:
That makes sense to me as well. I'll drive it back through API review in either case and will see which is preferred (`AllocZeroed` or `AllocZeroInitialized`)

Reply:
API review has said they are fine with the renames (baring anyone that objects in the next 24h).